apis/externalsecrets/v1alpha1/secretstore_vault_types.go

```diff
@@ -161,3 +168,20 @@ type VaultJwtAuth struct {
 	// authenticate with Vault using the JWT/OIDC authentication method
 	SecretRef esmeta.SecretKeySelector `json:"secretRef,omitempty"`
 }
+
+// VaultJwtAuth authenticates with Vault using the JWT/OIDC authentication
+// method, with the role name and token stored in a Kubernetes Secret resource.
+type VaultCertAuth struct {
+	// ClientCert is a certificate to authenticate using the Cert Vault
+	// authentication method
+	// +optional
+	ClientCert string `json:"clientCert"`
```

Comment:
The client cert is not sensitive information, however we could pull that out of a `Kind=Secret` aswell like the private key. Ideally a user has a `type=tls` secret. wdyt?

Reply:
Good idea